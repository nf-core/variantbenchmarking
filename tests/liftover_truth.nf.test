nextflow_pipeline {

    name "Test pipeline for liftover small germline variants, tested benchmarking methods are happy and rtgtools"
    script "../main.nf"
    tag "pipeline"
    tag "small"
    tag "germline"
    tag "liftover"
    tag "hg38"
    config "../conf/tests/liftover_truth.config"

    test("Params: --analysis 'germline' --variant_type 'small' --method 'happy,rtgtools' --liftover 'truth'") {

        when {
            params {
                outdir               = "$outputDir"
                test_data_base       = 'https://raw.githubusercontent.com/nf-core/test-datasets/variantbenchmarking'

            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
