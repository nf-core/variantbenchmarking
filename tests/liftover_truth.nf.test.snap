{
    "Params: --analysis 'germline' --variant_type 'small' --method 'happy,rtgtools' --liftover 'truth'": {
        "content": [
            68,
            null,
            [
                "pipeline_info",
                "pipeline_info/nf_core_variantbenchmarking_software_mqc_versions.yml",
                "references",
                "references/dictionary",
                "references/dictionary/genome.dict",
                "references/rtgtools",
                "references/rtgtools/genome.sdf",
                "references/rtgtools/genome.sdf/done",
                "references/rtgtools/genome.sdf/format.log",
                "references/rtgtools/genome.sdf/mainIndex",
                "references/rtgtools/genome.sdf/nameIndex0",
                "references/rtgtools/genome.sdf/namedata0",
                "references/rtgtools/genome.sdf/namepointer0",
                "references/rtgtools/genome.sdf/progress",
                "references/rtgtools/genome.sdf/reference.txt",
                "references/rtgtools/genome.sdf/seqdata0",
                "references/rtgtools/genome.sdf/seqdata1",
                "references/rtgtools/genome.sdf/seqdata2",
                "references/rtgtools/genome.sdf/seqdata3",
                "references/rtgtools/genome.sdf/seqpointer0",
                "references/rtgtools/genome.sdf/seqpointer1",
                "references/rtgtools/genome.sdf/seqpointer2",
                "references/rtgtools/genome.sdf/seqpointer3",
                "references/rtgtools/genome.sdf/sequenceIndex0",
                "references/rtgtools/genome.sdf/suffixIndex0",
                "references/rtgtools/genome.sdf/suffixdata0",
                "references/rtgtools/genome.sdf/suffixpointer0",
                "references/rtgtools/genome.sdf/summary.txt",
                "small",
                "small/HG002",
                "small/HG002/preprocess",
                "small/HG002/preprocess/HG002.vcf.gz",
                "small/HG002/preprocess/HG002.vcf.gz.tbi",
                "small/HG002/preprocess/liftover",
                "small/HG002/preprocess/liftover/HG002.reformatted.renamechr.vcf.gz",
                "small/HG002/preprocess/liftover/HG002_GRCh37_1_22_v4.2.1_highconf.liftedsort.merged.bed",
                "small/HG002/stats",
                "small/HG002/stats/bcftools",
                "small/HG002/stats/bcftools/HG002.bcftools_stats.txt",
                "small/multiqc",
                "small/multiqc/multiqc_data",
                "small/multiqc/multiqc_data/bcftools-stats-subtypes.txt",
                "small/multiqc/multiqc_data/bcftools_stats_indel-lengths.txt",
                "small/multiqc/multiqc_data/bcftools_stats_variant_depths.txt",
                "small/multiqc/multiqc_data/bcftools_stats_vqc_Count_Indels.txt",
                "small/multiqc/multiqc_data/bcftools_stats_vqc_Count_SNP.txt",
                "small/multiqc/multiqc_data/bcftools_stats_vqc_Count_Transitions.txt",
                "small/multiqc/multiqc_data/bcftools_stats_vqc_Count_Transversions.txt",
                "small/multiqc/multiqc_data/multiqc.log",
                "small/multiqc/multiqc_data/multiqc_bcftools_stats.txt",
                "small/multiqc/multiqc_data/multiqc_citations.txt",
                "small/multiqc/multiqc_data/multiqc_data.json",
                "small/multiqc/multiqc_data/multiqc_general_stats.txt",
                "small/multiqc/multiqc_data/multiqc_software_versions.txt",
                "small/multiqc/multiqc_data/multiqc_sources.txt",
                "small/multiqc/multiqc_plots",
                "small/multiqc/multiqc_plots/pdf",
                "small/multiqc/multiqc_plots/pdf/bcftools-stats-subtypes-cnt.pdf",
                "small/multiqc/multiqc_plots/pdf/bcftools-stats-subtypes-pct.pdf",
                "small/multiqc/multiqc_plots/pdf/bcftools_stats_indel-lengths.pdf",
                "small/multiqc/multiqc_plots/pdf/bcftools_stats_variant_depths.pdf",
                "small/multiqc/multiqc_plots/pdf/bcftools_stats_vqc_Count_Indels.pdf",
                "small/multiqc/multiqc_plots/pdf/bcftools_stats_vqc_Count_SNP.pdf",
                "small/multiqc/multiqc_plots/pdf/bcftools_stats_vqc_Count_Transitions.pdf",
                "small/multiqc/multiqc_plots/pdf/bcftools_stats_vqc_Count_Transversions.pdf",
                "small/multiqc/multiqc_plots/pdf/general_stats_table.pdf",
                "small/multiqc/multiqc_plots/png",
                "small/multiqc/multiqc_plots/png/bcftools-stats-subtypes-cnt.png",
                "small/multiqc/multiqc_plots/png/bcftools-stats-subtypes-pct.png",
                "small/multiqc/multiqc_plots/png/bcftools_stats_indel-lengths.png",
                "small/multiqc/multiqc_plots/png/bcftools_stats_variant_depths.png",
                "small/multiqc/multiqc_plots/png/bcftools_stats_vqc_Count_Indels.png",
                "small/multiqc/multiqc_plots/png/bcftools_stats_vqc_Count_SNP.png",
                "small/multiqc/multiqc_plots/png/bcftools_stats_vqc_Count_Transitions.png",
                "small/multiqc/multiqc_plots/png/bcftools_stats_vqc_Count_Transversions.png",
                "small/multiqc/multiqc_plots/png/general_stats_table.png",
                "small/multiqc/multiqc_plots/svg",
                "small/multiqc/multiqc_plots/svg/bcftools-stats-subtypes-cnt.svg",
                "small/multiqc/multiqc_plots/svg/bcftools-stats-subtypes-pct.svg",
                "small/multiqc/multiqc_plots/svg/bcftools_stats_indel-lengths.svg",
                "small/multiqc/multiqc_plots/svg/bcftools_stats_variant_depths.svg",
                "small/multiqc/multiqc_plots/svg/bcftools_stats_vqc_Count_Indels.svg",
                "small/multiqc/multiqc_plots/svg/bcftools_stats_vqc_Count_SNP.svg",
                "small/multiqc/multiqc_plots/svg/bcftools_stats_vqc_Count_Transitions.svg",
                "small/multiqc/multiqc_plots/svg/bcftools_stats_vqc_Count_Transversions.svg",
                "small/multiqc/multiqc_plots/svg/general_stats_table.svg",
                "small/multiqc/multiqc_report.html",
                "small/summary",
                "small/summary/comparisons",
                "small/summary/comparisons/rtgtools.FN.csv",
                "small/summary/comparisons/rtgtools.FP.csv",
                "small/summary/comparisons/rtgtools.TP_base.csv",
                "small/summary/comparisons/rtgtools.TP_comp.csv",
                "small/summary/datavzrd",
                "small/summary/datavzrd/happy",
                "small/summary/datavzrd/happy/index.html",
                "small/summary/datavzrd/happy/static",
                "small/summary/datavzrd/happy/static/bootstrap-select.min.css",
                "small/summary/datavzrd/happy/static/bootstrap-table-fixed-columns.min.css",
                "small/summary/datavzrd/happy/static/bootstrap-table.min.css",
                "small/summary/datavzrd/happy/static/bootstrap.min.css",
                "small/summary/datavzrd/happy/static/bundle.js",
                "small/summary/datavzrd/happy/static/datavzrd.css",
                "small/summary/datavzrd/happy/test",
                "small/summary/datavzrd/happy/test/config.js",
                "small/summary/datavzrd/happy/test/data",
                "small/summary/datavzrd/happy/test/data/data_1.js",
                "small/summary/datavzrd/happy/test/functions.js",
                "small/summary/datavzrd/happy/test/heatmap.js",
                "small/summary/datavzrd/happy/test/index_1.html",
                "small/summary/datavzrd/happy/test/plots",
                "small/summary/datavzrd/happy/test/plots/plot_0.js",
                "small/summary/datavzrd/happy/test/plots/plot_1.js",
                "small/summary/datavzrd/happy/test/plots/plot_10.js",
                "small/summary/datavzrd/happy/test/plots/plot_11.js",
                "small/summary/datavzrd/happy/test/plots/plot_12.js",
                "small/summary/datavzrd/happy/test/plots/plot_13.js",
                "small/summary/datavzrd/happy/test/plots/plot_14.js",
                "small/summary/datavzrd/happy/test/plots/plot_15.js",
                "small/summary/datavzrd/happy/test/plots/plot_16.js",
                "small/summary/datavzrd/happy/test/plots/plot_17.js",
                "small/summary/datavzrd/happy/test/plots/plot_18.js",
                "small/summary/datavzrd/happy/test/plots/plot_2.js",
                "small/summary/datavzrd/happy/test/plots/plot_3.js",
                "small/summary/datavzrd/happy/test/plots/plot_4.js",
                "small/summary/datavzrd/happy/test/plots/plot_5.js",
                "small/summary/datavzrd/happy/test/plots/plot_6.js",
                "small/summary/datavzrd/happy/test/plots/plot_7.js",
                "small/summary/datavzrd/happy/test/plots/plot_8.js",
                "small/summary/datavzrd/happy/test/plots/plot_9.js",
                "small/summary/datavzrd/rtgtools",
                "small/summary/datavzrd/rtgtools/index.html",
                "small/summary/datavzrd/rtgtools/static",
                "small/summary/datavzrd/rtgtools/static/bootstrap-select.min.css",
                "small/summary/datavzrd/rtgtools/static/bootstrap-table-fixed-columns.min.css",
                "small/summary/datavzrd/rtgtools/static/bootstrap-table.min.css",
                "small/summary/datavzrd/rtgtools/static/bootstrap.min.css",
                "small/summary/datavzrd/rtgtools/static/bundle.js",
                "small/summary/datavzrd/rtgtools/static/datavzrd.css",
                "small/summary/datavzrd/rtgtools/test",
                "small/summary/datavzrd/rtgtools/test/config.js",
                "small/summary/datavzrd/rtgtools/test/data",
                "small/summary/datavzrd/rtgtools/test/data/data_1.js",
                "small/summary/datavzrd/rtgtools/test/functions.js",
                "small/summary/datavzrd/rtgtools/test/heatmap.js",
                "small/summary/datavzrd/rtgtools/test/index_1.html",
                "small/summary/datavzrd/rtgtools/test/plots",
                "small/summary/datavzrd/rtgtools/test/plots/plot_0.js",
                "small/summary/datavzrd/rtgtools/test/plots/plot_1.js",
                "small/summary/datavzrd/rtgtools/test/plots/plot_2.js",
                "small/summary/datavzrd/rtgtools/test/plots/plot_3.js",
                "small/summary/datavzrd/rtgtools/test/plots/plot_4.js",
                "small/summary/datavzrd/rtgtools/test/plots/plot_5.js",
                "small/summary/datavzrd/rtgtools/test/plots/plot_6.js",
                "small/summary/datavzrd/rtgtools/test/plots/plot_7.js",
                "small/summary/datavzrd/rtgtools/test/plots/plot_8.js",
                "small/summary/datavzrd/versions.yml",
                "small/summary/plots",
                "small/summary/plots/happy",
                "small/summary/plots/happy/INDEL_ALL_metric_by_tool_happy_mqc.png",
                "small/summary/plots/happy/INDEL_ALL_variants_by_tool_happy_mqc.png",
                "small/summary/plots/happy/INDEL_PASS_metric_by_tool_happy_mqc.png",
                "small/summary/plots/happy/INDEL_PASS_variants_by_tool_happy_mqc.png",
                "small/summary/plots/happy/SNP_ALL_metric_by_tool_happy_mqc.png",
                "small/summary/plots/happy/SNP_ALL_variants_by_tool_happy_mqc.png",
                "small/summary/plots/happy/SNP_PASS_metric_by_tool_happy_mqc.png",
                "small/summary/plots/happy/SNP_PASS_variants_by_tool_happy_mqc.png",
                "small/summary/plots/rtgtools",
                "small/summary/plots/rtgtools/metric_by_tool_rtgtools_mqc.png",
                "small/summary/plots/rtgtools/variants_by_tool_rtgtools_mqc.png",
                "small/summary/tables",
                "small/summary/tables/happy",
                "small/summary/tables/happy/happy.summary.csv",
                "small/summary/tables/rtgtools",
                "small/summary/tables/rtgtools/rtgtools.summary.csv",
                "small/test1",
                "small/test1/benchmarks",
                "small/test1/benchmarks/happy",
                "small/test1/benchmarks/happy/test1.HG002.strelka.extended.csv",
                "small/test1/benchmarks/happy/test1.HG002.strelka.metrics.json.gz",
                "small/test1/benchmarks/happy/test1.HG002.strelka.roc.Locations.INDEL.PASS.csv.gz",
                "small/test1/benchmarks/happy/test1.HG002.strelka.roc.Locations.INDEL.csv.gz",
                "small/test1/benchmarks/happy/test1.HG002.strelka.roc.Locations.SNP.PASS.csv.gz",
                "small/test1/benchmarks/happy/test1.HG002.strelka.roc.Locations.SNP.csv.gz",
                "small/test1/benchmarks/happy/test1.HG002.strelka.roc.all.csv.gz",
                "small/test1/benchmarks/happy/test1.HG002.strelka.runinfo.json",
                "small/test1/benchmarks/happy/test1.HG002.strelka.summary.csv",
                "small/test1/benchmarks/happy/test1.HG002.strelka.vcf.gz",
                "small/test1/benchmarks/happy/test1.HG002.strelka.vcf.gz.tbi",
                "small/test1/benchmarks/rtgtools",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.fn.vcf.gz",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.fn.vcf.gz.tbi",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.fp.vcf.gz",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.fp.vcf.gz.tbi",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.non_snp_roc.tsv.gz",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.phasing.txt",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.snp_roc.tsv.gz",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.summary.txt",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.tp-baseline.vcf.gz",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.tp-baseline.vcf.gz.tbi",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.tp.vcf.gz",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.tp.vcf.gz.tbi",
                "small/test1/benchmarks/rtgtools/test1.HG002.strelka.weighted_roc.tsv.gz",
                "small/test1/preprocess",
                "small/test1/preprocess/test1.vcf.gz",
                "small/test1/preprocess/test1.vcf.gz.tbi",
                "small/test1/stats",
                "small/test1/stats/bcftools",
                "small/test1/stats/bcftools/test1.strelka.bcftools_stats.txt",
                "small/test2",
                "small/test2/benchmarks",
                "small/test2/benchmarks/happy",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.extended.csv",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.metrics.json.gz",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.roc.Locations.INDEL.PASS.csv.gz",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.roc.Locations.INDEL.csv.gz",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.roc.Locations.SNP.PASS.csv.gz",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.roc.Locations.SNP.csv.gz",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.roc.all.csv.gz",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.runinfo.json",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.summary.csv",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.vcf.gz",
                "small/test2/benchmarks/happy/test2.HG002.bcftools.vcf.gz.tbi",
                "small/test2/benchmarks/rtgtools",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.fn.vcf.gz",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.fn.vcf.gz.tbi",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.fp.vcf.gz",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.fp.vcf.gz.tbi",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.non_snp_roc.tsv.gz",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.phasing.txt",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.snp_roc.tsv.gz",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.summary.txt",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.tp-baseline.vcf.gz",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.tp-baseline.vcf.gz.tbi",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.tp.vcf.gz",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.tp.vcf.gz.tbi",
                "small/test2/benchmarks/rtgtools/test2.HG002.bcftools.weighted_roc.tsv.gz",
                "small/test2/preprocess",
                "small/test2/preprocess/test2.vcf.gz",
                "small/test2/preprocess/test2.vcf.gz.tbi",
                "small/test2/stats",
                "small/test2/stats/bcftools",
                "small/test2/stats/bcftools/test2.bcftools.bcftools_stats.txt"
            ],
            [
                "nameIndex0:md5,720ad6dfad3d1ee07a3a2bc3ed953395",
                "namedata0:md5,758c4a60948aed6c6a4c78323aa73ab3",
                "namepointer0:md5,319f9f665b338f0e218677e036637e94",
                "reference.txt:md5,c9f7ef637449eb0f006d90080a130e30",
                "seqdata0:md5,ea8728adf43db158305ba188de66e838",
                "seqdata1:md5,d3470516d52b20e9477323d051846c3f",
                "seqdata2:md5,1fd1a98f39b966aac0b696b122a1c6d5",
                "seqdata3:md5,49ba3e2d1b4ada1d7ed2c29079833491",
                "seqpointer0:md5,77ad13d617e120eadad10bdec41cc00f",
                "seqpointer1:md5,fce3fdba9b237ec51f3c356086d0f8b9",
                "seqpointer2:md5,3d99bc0313f17f491a0a2903d6585b14",
                "seqpointer3:md5,2a69114e099aa0752e78f417cc8fe7bb",
                "sequenceIndex0:md5,18c7d37abf02d389fda5d9945b16b614",
                "suffixIndex0:md5,b7bb2ba061ab54c0bf07c0a941d0277a",
                "suffixdata0:md5,f2876dd730673cd49c4de191001f634e",
                "suffixpointer0:md5,468281ffb10d7dd934289af762a03781",
                "HG002_GRCh37_1_22_v4.2.1_highconf.liftedsort.merged.bed:md5,5e997133249c2227cb5363b314245636",
                "HG002.bcftools_stats.txt:md5,db248c6e7e92253abd836c1d8a326371",
                "test1.HG002.strelka.extended.csv:md5,fadf06276179c1cc388007e5651fb9af",
                "test1.HG002.strelka.roc.Locations.INDEL.PASS.csv.gz:md5,a08665be8ea20855a14d6418d07521dc",
                "test1.HG002.strelka.roc.Locations.INDEL.csv.gz:md5,8d4a20a5914379230952aeb252845e2d",
                "test1.HG002.strelka.roc.Locations.SNP.PASS.csv.gz:md5,c546609a3e23b4c5c97ee0fc2e864e07",
                "test1.HG002.strelka.roc.Locations.SNP.csv.gz:md5,8d2600c17665ed1299e2dd483705ddaa",
                "test1.HG002.strelka.roc.all.csv.gz:md5,d0a4408438b2182cb82caec918c090f1",
                "test1.HG002.strelka.summary.csv:md5,001fa2371777d1d2dc3eb0f4dcaca197",
                "test1.HG002.strelka.phasing.txt:md5,0c9340e64e032b5f74d0ea3773464afd",
                "test1.HG002.strelka.summary.txt:md5,4b775edb878759727ef66d5d5b2b9ee8",
                "test1.strelka.bcftools_stats.txt:md5,869e7eac4d1bde15c33ac3598af3b58e",
                "test2.HG002.bcftools.extended.csv:md5,05c6efd85a9b9823a5ddfe6619be3323",
                "test2.HG002.bcftools.roc.Locations.INDEL.PASS.csv.gz:md5,463cee548b7bdf133c0ae9b5803a50ca",
                "test2.HG002.bcftools.roc.Locations.INDEL.csv.gz:md5,424e39c327ae6ef9001942bc895abdc9",
                "test2.HG002.bcftools.roc.Locations.SNP.PASS.csv.gz:md5,508277f9aa9a541812360bbf914ae3fc",
                "test2.HG002.bcftools.roc.Locations.SNP.csv.gz:md5,87a524b9f5eefdd4d300e696b21948a2",
                "test2.HG002.bcftools.roc.all.csv.gz:md5,73a59d38316a913e6ec445ef4efec207",
                "test2.HG002.bcftools.summary.csv:md5,38f3d8eb32c4d006a5623777ad265c39",
                "test2.HG002.bcftools.phasing.txt:md5,38920536b8c3e241e873c07ba61762e6",
                "test2.HG002.bcftools.summary.txt:md5,264eb2b064a9ee4b098470a13c2887c9",
                "test2.bcftools.bcftools_stats.txt:md5,2efc3e61c4aaa5bb2da48803c81d8fcc"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-29T17:04:40.563386602"
    }
}